stock_move_backdating/wizards/stock_transfer_details.py

```diff
@@ -40,6 +40,14 @@ def default_get(self, fields_list):
             packop['date'] = now
         return res
 
+    @api.one
+    def do_detailed_transfer(self):
+        ctx = self.env.context.copy()
+        ctx.update({'move_date': self.date_backdating})
+        res = super(StockTransferDetails, self.with_context(
```

Comment:
You could simply pass 'move_date' by keyword without copy the orginal context.
See https://www.odoo.com/documentation/8.0/reference/orm.html#altering-the-environment as reference.